src/util/GeoSparqlHelpers.h

```diff
@@ -0,0 +1,107 @@
+//
+// }  // namespace ad_utility
+
+// Helper function that checks whether a string looks like a WKT pointC (just
+// checks the beginning and that it has the minimal length).
+namespace {
+bool is_wkt_point(const std::string& p) {
+  return p.size() >= 12 && p[0] == '"' && (p[1] == 'P' || p[1] == 'p') &&
+         (p[2] == 'O' || p[2] == 'o') && (p[3] == 'I' || p[3] == 'i') &&
+         (p[4] == 'N' || p[4] == 'n') && (p[5] == 'T' || p[5] == 't') &&
+         p[6] == '(';
+}
+
+// Parse a single WKT Point of the form "POINT(<double> <double>)", where the
+// quote are included, the POINT must be uppercase, there is an arbitrary
+// amount of whitespace after the comma, and arbitrary contents after the
+// final quote. If the string does not match this format, a pair of
+// `quiet_NaN` is returned
+std::pair<double, double> parse_wkt_point(const std::string& p) {
+  auto error_result = std::pair(std::numeric_limits<double>::quiet_NaN(),
+                                std::numeric_limits<double>::quiet_NaN());
+  if (!is_wkt_point(p)) {
+    return error_result;
+  }
+  const char* beg = p.c_str() + 7;
+  char* end;
+  double lng = std::strtod(beg, &end);
+  if (end == nullptr) return error_result;
+  beg = end + 1;
+  double lat = std::strtod(beg, &end);
+  if (end == nullptr || *end != ')' || *(end + 1) != '"') return error_result;
+  return std::pair(lat, lng);
+}
+
+}  // namespace
+
+namespace ad_utility {
+
+// Parse the longitude coordinate from a WKT point (it's the first coordinate).
+inline auto wktLongitude = [](const std::string& p) -> double {
+  auto nan = std::numeric_limits<double>::quiet_NaN();
+  if (!is_wkt_point(p)) return nan;
+  const char* beg = p.c_str() + 7;
+  char* end;
+  double lng = std::strtod(beg, &end);
+  return end != nullptr || *(end + 1) == ' ' ? lng : nan;
+};
+
+// Parse the latitute coordinate from a WKT point (it's the second coordinate).
+inline auto wktLatitude = [](const std::string& p) -> double {
+  auto nan = std::numeric_limits<double>::quiet_NaN();
+  if (!is_wkt_point(p)) return nan;
+  const char* beg = p.c_str() + 7;
+  while (*beg == ' ') ++beg;
+  if (*beg == '-') ++beg;
+  while (isdigit(*beg) || *beg == '.') ++beg;
+  char* end;
+  double lat = std::strtod(beg, &end);
+  return end != nullptr || *(end + 1) == ')' ? lat : nan;
+};
+
+// Compute the distance in km between two WKT points according to the formula in
+// https://en.wikipedia.org/wiki/Geographical_distance ("ellipsoidal earth
+// projected to a plane"). A more precise way is the Haversine formula, which we
+// save for when we compute this at indexing time.
+inline auto wktDist = [](const std::string& p1,
+                         const std::string& p2) -> double {
+  auto latlng1 = parse_wkt_point(p1);
+  auto latlng2 = parse_wkt_point(p2);
+  auto sqr = [](double x) { return x * x; };
+  auto m = M_PI / 180.0 * (latlng1.first + latlng2.first) / 2.0;
```

Comment:
Since C++20 we have proper standardized PI constants:
[https://godbolt.org/z/zdcq5Ye6q](https://godbolt.org/z/zdcq5Ye6q)

Reply:
(they work with all our supported compilers)

Reply:
Thanks, using std::numbers::pi now!